packages/camera/camera_android_camerax/lib/src/camera_selector.dart

```diff
@@ -0,0 +1,159 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+
+import 'android_camera_camerax_flutter_api_impls.dart';
+import 'camera_info.dart';
+import 'camerax_library.pigeon.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+
+/// Selects a camera for use.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/CameraSelector.
+class CameraSelector extends JavaObject {
+  /// Creates a [CameraSelector].
+  CameraSelector.detached(
+      {this.binaryMessenger, this.instanceManager, this.lensFacing})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = CameraSelectorHostApiImpl(
+      binaryMessenger: binaryMessenger,
+      instanceManager: instanceManager,
+    );
+    AndroidCameraXCameraFlutterApis.instance.ensureSetUp();
+  }
+
+  late final CameraSelectorHostApiImpl _api;
+
+  /// ID for front facing lens.
+  static const int LENS_FACING_FRONT = 0;
+
+  /// ID for back facing lens.
+  static const int LENS_FACING_BACK = 1;
+
+  /// Selector for default front facing camera.
+  static final Future<CameraSelector> defaultFrontCamera =
```

Comment:
I don't know if this is guaranteed to be correct that the Java:

```java
DEFAULT_BACK_CAMERA == new CameraSelector.Builder().requireLensFacing(LENS_FACING_BACK).build()
```

 I think this method could use the `attach` pattern from the `Wrapping Native Api`s doc. It would also make it synchronous. e.g.

```dart
static CameraSelector defaultFrontCamera({
  BinaryMessenger? binaryMessenger,
  InstanceManager? instanceManager,
}) {
  final CameraSelector cameraSelector = CameraSelector.detached();
  CameraSelectorHostApiImpl(
    binaryMessenger: binaryMessenger,
    instanceManager: instanceManager,
  ).attachDefaultFrontCameraFromInstances(cameraSelector);
  return cameraSelector;
}
```

Reply:
I confirmed they are equivalent: https://android.googlesource.com/platform/frameworks/support/+/f2e05c341382db64d127118a13451dcaa554b702/camera/camera-core/src/main/java/androidx/camera/core/CameraSelector.java#47